cloudstack/resources.go

```diff
@@ -149,3 +149,18 @@ func importStatePassthrough(d *schema.ResourceData, meta interface{}) ([]*schema
 
 	return []*schema.ResourceData{d}, nil
 }
+
+type zoneIDSetter interface {
+	SetZoneid(string)
+}
+
+func withZoneID(zoneid string) cloudstack.OptionFunc {
+	return func(_ *cloudstack.CloudStackClient, p interface{}) error {
+		zs, ok := p.(zoneIDSetter)
+		if !ok || zoneid == "" {
+			return nil
+		}
+		zs.SetZoneid(zoneid)
+		return nil
+	}
+}
```

Comment:
While this solves the problem, it would probably be better to place this in the `go-cloudstack` client. Are you up for making that change, or should I? Please note that it should be added in the `generate` sub package, as all the code in `go-cloudstack` is generated.

There already are a few other `WithXXX` functions in there that can serve as examples...

Reply:
Sure, this func is a nice fit for go-cloudstack. I'll send a PR .